src/components/TodoHeader/TodoHeader.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { USER_ID } from '../../utils/userId';
+import { addTodo, updateTodo } from '../../api/todos';
+import { TodoContext } from '../TodoContext';
+import { ErrorContext } from '../ErrorContext';
+import { GlobalLoader } from '../../types/GlobalLoader';
+
+type Props = {
+  onTempTodoAdd: (todo: Todo | null) => void;
+  tempTodo: Todo | null;
+  onGlobalLoaderChange: (globalLoader: GlobalLoader) => void;
+};
+
+export const TodoHeader: React.FC<Props> = (props) => {
+  const {
+    onTempTodoAdd,
+    tempTodo,
+    onGlobalLoaderChange,
+  } = props;
+
+  const { setError } = useContext(ErrorContext);
+  const { todos, setTodos } = useContext(TodoContext);
+  const [title, setTitle] = useState('');
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const isAllCompleted = todos.every(todo => todo.completed);
+
+  const handleSubmitForm = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title) {
+      setError('Title can\'t be empty');
+
+      return;
+    }
+
+    onTempTodoAdd({
+      id: 0,
+      userId: USER_ID,
+      title,
+      completed: false,
+    });
+  };
+
+  const handleToggleAll = () => {
+    const updatedTodos: Promise<Todo>[] = [];
+
+    if (isAllCompleted) {
+      onGlobalLoaderChange(GlobalLoader.Completed);
+      todos.forEach(todo => {
+        updatedTodos.push(updateTodo(todo.id, { completed: false })
+          .then(updatedTodo => updatedTodo)
+          .catch((error) => {
+            throw error;
+          }));
+      });
+    } else {
+      onGlobalLoaderChange(GlobalLoader.Active);
+      todos.forEach(todo => {
+        updatedTodos.push(updateTodo(todo.id, { completed: true })
+          .then(updatedTodo => updatedTodo)
+          .catch((error) => {
+            throw error;
+          }));
+      });
+    }
```

Comment:
Your actions differ only in status for `completed`, you don't need to completely rewrite the logic for this:
```suggestion
    onGlobalLoaderChange(isAllCompleted ? GlobalLoader.Completed : GlobalLoader.Active);
    todos.forEach(todo => {
      updateTodo(todo.id, { completed: !isAllCompleted })
        .then(updatedTodo => updatedTodos.push(updatedTodo))
        .catch((error) => throw error);
    });
```